### PR TITLE
feat(ui): Add custom alert bar

### PR DESCRIPTION
**Pull request by Hemanthghs**
Part of #947

## Summary by CodeRabbit

- **New Features**
	- Added a `SnackBar` component to display error messages dynamically based on application state.

- **Enhancements**
	- Integrated the `SnackBar` into the `SideBar` component for improved user feedback on errors.
	- Introduced a new constant `ALERT_HIDE_DURATION` to control the duration of error message display.

### CODE REVIEW

File: frontend/src/components/CustomAlertBar.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { useAppSelector } from '@/custom-hooks/StateHooks';
```
Comment (charymalloju) [inline frontend/src/components/CustomAlertBar.tsx:1]: Did you create this alert bar only for error messages? If so, please rename the file to CustomErrorAlert.

Comment (Hemanthghs) [reply frontend/src/components/CustomAlertBar.tsx:1]: It is both for error message and success message

Comment (aleem1314) [reply frontend/src/components/CustomAlertBar.tsx:1]: Why not name it `Snackbar`?

Comment (Hemanthghs) [reply frontend/src/components/CustomAlertBar.tsx]: Changed the name to SnackBar

Comment (charymalloju) [reply frontend/src/components/CustomAlertBar.tsx]: We already have a library for Snackbar, but we can add a prefix to avoid confusion.

**Review by charymalloju — APPROVED** [review]

**Pull request merged**